File: python/example/test-restart-client.py
# Template Setup
import riffle
from riffle import want

riffle.SetFabricLocal()
riffle.SetLogLevelDebug()

class GenericDomain(riffle.Domain):

    def onJoin(self):
        # End Template Setup
        ######################################################################################
        # Example Test Restart before call - Does restarting before a call work?
        import time
        # Trying to restart the node before the call happens, but need to insert artificial delay to make this happen!
        print "___NODERESTART___"
        time.sleep(0.5)
        s = backend.call("restartBeforeC", "Restart before call").wait(str)
        print(s) # Expects a str, like "Restart before call works"
        # End Example Test Restart before call

        print "___SETUPCOMPLETE___"

# Template Setup
app = riffle.Domain("xs.demo.test") # ARBITER $DOMAIN replaces "xs.demo.test"

client = riffle.Domain("client", superdomain=app)
backend = riffle.Domain("backend", superdomain=app)

GenericDomain("client", superdomain=app).join() # ARBITER $SUBDOMAIN replaces "client"
# End Template Setup

File: js/example/test-restart-client.js
var riffle = require('jsriffle');

riffle.setFabricLocal();
riffle.setLogLevelDebug();

var app = riffle.Domain("xs.demo.test");
var backend = app.subdomain("backend");
var client = app.subdomain("client");

client.onJoin = function() {
    /////////////////////////////////////////////////////////////////////////////////////
    // Example Test Restart before call - Does restarting before a call work?
    // Trying to restart the node before the call happens, but need to insert
    // artificial delay to make this happen!
    console.log("___NODERESTART___");
    setTimeout(function (){
        backend.call("restartBeforeC", "Restart before call").want(String).then(function(s) {
            console.log(s); // Expects a String, like "Restart before call works"
        });
    }, 6000);
    // End Example Test Restart before call

    /////////////////////////////////////////////////////////////////////////////////////
    // Example Test Restart after reg - Does restarting after a register work
    setTimeout(function (){
        backend.call("restartAfterR", "Restart after reg").want(String).then(function(s) {
            console.log(s); // Expects a String, like "Restart after reg works"
        });
    }, 4000);
    // End Example Test Restart after reg

    console.log("___SETUPCOMPLETE___");
};

client.join()
